util/failover.go

```diff
@@ -316,3 +302,69 @@ func GetPreferredNodes(clientset *kubernetes.Clientset, selector, namespace stri
 
 	return nodes, err
 }
+
+// ToggleAutoFailover enables or disables autofailover for a cluster.  Disabling autofailover means "pausing"
+// Patroni, which will result in Patroni stepping aside from managing the cluster.  This will effectively cause
+// Patroni to stop responding to failures or other database activities, e.g. it will not attempt to start the
+// database when stopped to perform maintenance
+func ToggleAutoFailover(clientset *kubernetes.Clientset, enable bool, clusterName, namespace string) error  {
+	
+	// find the "config" configMap created by Patroni
+	configMapName := clusterName+"-config"
+	log.Debugf("setting autofailover to %t for cluster %s", enable, clusterName)
+
+	configMap, found := kubeapi.GetConfigMap(clientset, configMapName, namespace)
+	if found {
+		configJSONStr := configMap.ObjectMeta.Annotations["config"]
+
+		var configJSON map[string]interface{}
+		json.Unmarshal([]byte(configJSONStr), &configJSON)
+
+		if !enable {  // disable autofail condition
+			// If "pause" isn't present in the config then assume autofail is enabled and needs to be disabled.  Or if
+			// if it is present and set to "false", the is is also enabled an needs to be disabled."
+			if _,ok := configJSON["pause"] ; !ok || configJSON["pause"] == false  {
+				log.Debugf("updating pause key in configMap %s to disable autofailover", configMapName)
+				// disable autofail by setting "pause" to true 
+				configJSON["pause"] = true
+				configJSONFinalStr, err := json.Marshal(configJSON)
+				if err != nil {
+					return err
+				}
+				configMap.ObjectMeta.Annotations["config"] = string(configJSONFinalStr)
+				err = kubeapi.UpdateConfigMap(clientset, configMap, namespace)
+				if err != nil {
+					return err
+				}
+			} else {
+				log.Debugf("autofailover already disabled according to the pause key in configMap %s", 
+					configMapName)
+			}
+		}  else {  // enable autofail condition
+			// If "pause" is present in the config and set to "true", then it needs to be removed.  Otherwise, if
+			// if "pause" isn't present in the config or if it has a value other than true, then assume autofail 
+			// is enabled and do nothing (when Patroni see's an invalid value for "pause" it sets it to "true")
+			if _,ok := configJSON["pause"] ; ok || configJSON["pause"] == true  {
+				//  disabled autofail by removing "pause" from the config
+				delete(configJSON, "pause")
+				configJSONFinalStr, err := json.Marshal(configJSON)
+				if err != nil {
+					return err
+				}
+				configMap.ObjectMeta.Annotations["config"] = string(configJSONFinalStr)
+				err = kubeapi.UpdateConfigMap(clientset, configMap, namespace)
+				if err != nil {
+					return err
+				}
+			} else {
+				log.Debugf("autofailover already enabled according to the pause key (or lack thereof) in configMap %s", 
+					configMapName)
+			}
+		}
```

Comment:
I'd recommend breaking these two clauses out into `disableAutofail` and `enableAutofail` helper functions.